tests/integration/kubernetes/k8s-guest-pull-image-authenticated.bats

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    # TODO - do TEEs want to run these tests?
+    if is_confidential_hardware; then
+        skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
+    fi
+
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    setup_common
+    AUTHENTICATED_IMAGE="${AUTHENTICATED_IMAGE:-quay.io/kata-containers/confidential-containers-auth:test}"
+    AUTHENTICATED_IMAGE_USER=${AUTHENTICATED_IMAGE_USER:-}
+    AUTHENTICATED_IMAGE_PASSWORD=${AUTHENTICATED_IMAGE_PASSWORD:-}
+
+    # Set up Kubernetes secret for the nydus-snapshotter metadata pull
+    kubectl delete secret cococred --ignore-not-found
+    kubectl create secret docker-registry cococred --docker-server="https://"$(echo "$AUTHENTICATED_IMAGE" | cut -d':' -f1) \
+    --docker-username="${AUTHENTICATED_IMAGE_USER}" --docker-password="${AUTHENTICATED_IMAGE_PASSWORD}"
+}
+
+function setup_kbs_credentials() {
+    image=$1
+    user=$2
+    password=$3
+
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    if [[ -z ${user} || -z ${password} ]]; then
+        if [[ -n ${GITHUB_ACTION} ]]; then
+            die "User and/or password not supplied to authenticated registry test"
+        else
+            skip "running local and missing user/password"
+        fi
+    fi
+
+    registry_credential_encoded=$(echo "${user}:${password}" | base64 -w 0)
+    registry=$(echo "$image" | cut -d':' -f1)
+
+    auth_json=$(echo "{
+        "auths": {
+            "${registry}": {
+                "auth": "${registry_credential_encoded}"
+            }
+        }
+    }" | base64 -w 0)
+
+    if ! is_confidential_hardware; then
+        kbs_set_allow_all_resources
+    fi
+
+    kbs_set_resource "default" "credentials" "test" "${auth_json}"
```

Comment:
Oh. The resource of registry auth should not be base64 encoded. should be plaintext. But original `kbs_set_resource` might not support multiple line content, thus we need similar handler as https://github.com/kata-containers/kata-containers/pull/9382/commits/7bfacf6a512ae5e3570f975289179a39c1468ce2#diff-12641f3199d1854db050dfb1eef70162bb8b41bef72db77ba0b35910907c72c1R43 did.

Reply:
Ah - that's probably what is wrong. Sorry - In the offline_fs_kbc we base64 encoded lots of things, so I assumed it was the same here. Let's see if we can fix that up...